pkg/kptfile/kptfileutil/util.go

```diff
@@ -173,6 +173,63 @@ func HasKptfile(path string) (bool, error) {
 	return true, nil
 }
 
+func MergeAndUpdateLocal(local, updated, original string) error {
+	// hasUpdatedKf := true
+	updatedKf, err := ReadFile(updated)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+		// hasUpdatedKf = false
+	}
+
+	// hasOriginalKf := true
+	originalKf, err := ReadFile(original)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+		// hasOriginalKf = false
+	}
+
+	localKf, err := ReadFile(local)
+	if err != nil {
+		return err
+	}
+
+	newKf, err := Merge(localKf, updatedKf, originalKf)
+	if err != nil {
+		return err
+	}
+
+	return WriteFile(local, newKf)
+}
+
+func Merge(local, updated, _ kptfilev1alpha2.KptFile) (kptfilev1alpha2.KptFile, error) {
```

Comment:
Why is merge not taking origin ? It should be 3-way merge correct ?

Reply:
For now, we only handle `upstream` and `upstreamLock` in the simplest way, so it is not yet needed. But when we add code for merging of the pipeline, we will definitely need it.